src/MatrixClientPeg.ts

```diff
@@ -273,7 +272,11 @@ class MatrixClientPegClass implements IMatrixClientPeg {
         opts.clientWellKnownPollPeriod = 2 * 60 * 60; // 2 hours
         opts.threadSupport = true;
 
-        opts.slidingSync = await SlidingSyncManager.instance.setup(this.matrixClient)
+        if (SettingsStore.getValue("feature_sliding_sync")) {
+            opts.slidingSync = await SlidingSyncManager.instance.setup();
+        } else {
+            await SlidingSyncManager.instance.checkSupport();
```

Comment:
I would either not await this or add a try-catch to not allow it to crash the app startup